waspc/data/Generator/templates/react-app/scripts/validate-env.mjs

```diff
@@ -0,0 +1,7 @@
+import { isValidAbsoluteURL } from './shared/validators.mjs';
+
+console.info("🔍 Validating environment variables...");
+if (process.env.REACT_APP_API_URL && !isValidAbsoluteURL(process.env.REACT_APP_API_URL)) {
+    throw 'Environemnt variable REACT_APP_API_URL is not a valid absolute URL';
+}
```

Comment:
This block is identical to the one in `server` (modulo env var name). Can we put it in `validators.mjs` as well? We could use it like `throwIfInvalidUrl(process.env.REACT_API_URL)`.

Btw, we're throwing strings. Aren't they usually wrapped in `Error`s? Is this an output thing?

Reply:
That's an interesting idea, there is some logic here (if the variable is set then check it) which would be good if it was extracted.

I dropped the `Error` wrapper to get a slightly nicer output IMHO. What do you think:
```js
// Wrapped with Error
Web app(stderr): file:///Users/ilakovac/wasp-lang/wasp/waspc/examples/todoApp/.wasp/out/web-app/scripts/validate-env.mjs:5
Web app(stderr):     throw new Error('Environemnt variable REACT_APP_API_URL is not a valid absolute URL');
Web app(stderr):           ^
Web app(stderr):
Web app(stderr): Error: Environemnt variable REACT_APP_API_URL is not a valid absolute URL
Web app(stderr):     at file:///Users/ilakovac/wasp-lang/wasp/waspc/examples/todoApp/.wasp/out/web-app/scripts/validate-env.mjs:5:11
Web app(stderr):     at ModuleJob.run (node:internal/modules/esm/module_job:193:25)
Web app(stderr):     at async Promise.all (index 0)
Web app(stderr):     at async ESMLoader.import (node:internal/modules/esm/loader:530:24)
Web app(stderr):     at async loadESM (node:internal/process/esm_loader:91:5)
Web app(stderr):     at async handleMainPromise (node:internal/modules/run_main:65:12)
Web app(stderr):
Web app(stderr): Node.js v18.12.1
```

```js
// Plain string
Web app(stderr): node:internal/process/esm_loader:97
Web app(stderr):     internalBinding('errors').triggerUncaughtException(
Web app(stderr):                               ^
Web app(stderr): Environemnt variable REACT_APP_API_URL is not a valid absolute URL
Web app(stderr): (Use `node --trace-uncaught ...` to show where the exception was thrown)
Web app(stderr):
Web app(stderr): Node.js v18.12.1
```

Now that I look at it side by side, it seems that the `Error` wrapped one provides a location of the throw, which can be useful.

Reply:
There's still some repetition here, why not just:
```
throwIfInvalidUrl(process.env.REACT_API_URL)
```
And report the message inside (it should be the same for all env vars).

Reply:
I wrote this function without thinking about `env` variables, that's why I wanted to pass in the full error message. 

I could reduce it to `throwIfInvalidUrl (value, name)` and put inside ```throw new Error(`${name} is not valid absolute URL`)``` 

Then we could send in `throwIfInvalidUrl(process.env.REACT_API_URL, 'Env variable REACT_API_URL')`

Reply:
I've updated it here: https://github.com/wasp-lang/wasp/pull/994/commits/e6000d2ffd8328c706d9d95cbb377c3c4db6361b